src/mysky/index.ts

```diff
@@ -376,14 +401,92 @@ export class MySky {
     const dataKey = deriveDiscoverableTweak(path);
     opts.hashedDataKeyHex = true; // Do not hash the tweak anymore.
 
-    const entry = await getDeletionRegistryEntry(this.connector.client, publicKey, dataKey, opts);
+    const getEntryOpts = extractOptions(opts, defaultGetEntryOptions);
+    const entry = await getNextRegistryEntry(
+      this.connector.client,
+      publicKey,
+      dataKey,
+      new Uint8Array(RAW_SKYLINK_SIZE),
+      getEntryOpts
+    );
 
     const signature = await this.signRegistryEntry(entry, path);
 
     const setEntryOpts = extractOptions(opts, defaultSetEntryOptions);
     await this.connector.client.registry.postSignedEntry(publicKey, entry, signature, setEntryOpts);
   }
 
+  /**
+   * Gets the raw registry entry data for the given path, if the user has given READ permissions.
+   *
+   * @param path - The data path.
+   * @param [customOptions] - Additional settings that can optionally be set.
+   * @returns - The entry data.
+   */
+  async getEntryData(path: string, customOptions: CustomGetEntryOptions): Promise<EntryData> {
+    validateString("path", path, "parameter");
+    validateOptionalObject("customOptions", customOptions, "parameter", defaultGetEntryOptions);
+
+    const opts = {
+      ...defaultGetEntryOptions,
+      ...this.connector.client.customOptions,
+      ...customOptions,
+    };
+
+    const publicKey = await this.userID();
+    const dataKey = deriveDiscoverableTweak(path);
+    opts.hashedDataKeyHex = true; // Do not hash the tweak anymore.
+
+    const { entry } = await this.connector.client.registry.getEntry(publicKey, dataKey, opts);
+    if (!entry) {
+      return { data: null };
+    }
+    return { data: entry.data };
+  }
+
+  /**
+   * Sets the entry data at the given path, if the user has given WRITE permissions.
+   *
+   * @param path - The data path.
+   * @param data - The raw entry data to set.
+   * @param [customOptions] - Additional settings that can optionally be set.
+   * @returns - The entry data.
+   * @throws - Will throw if the length of the data is > 70 bytes.
+   */
+  async setEntryData(path: string, data: Uint8Array, customOptions: CustomSetEntryOptions): Promise<EntryData> {
+    validateString("path", path, "parameter");
+    validateUint8Array("data", data, "parameter");
+    validateOptionalObject("customOptions", customOptions, "parameter", defaultGetEntryOptions);
+
+    if (data.length > MAX_ENTRY_LENGTH) {
+      throwValidationError(
+        "data",
+        data,
+        "parameter",
+        `'Uint8Array' of length <= ${MAX_ENTRY_LENGTH}, was length ${data.length}`
```

Comment:
(nit) kind of a confusing way to put it

Reply:
I'm not sure what you mean. The error will be something like `Expected to be 'Uint8Array' of length <= X, was length Y`. I'm open to suggestions though.